src/Service/AuthorizationService.php

```diff
@@ -98,45 +79,12 @@ public function isGranted(?IdentityInterface $identity, string $permission, $con
             return false;
         }
 
-        if ($this->hasAssertion($permission)) {
-            return $this->assert($this->assertions[$permission], $permission, $identity, $context);
-        }
+        if (! empty($this->assertions[$permission])) {
+            $assertionSet = new AssertionSet($this->assertionPluginManager, $this->assertions);
```

Comment:
yes, we will have to create one more object but is it that bad? But this will allow us to create nested assertions example:
````php
[
    'zfc_rbac' => [
        'assertion_map' => [
            // single assertion
            'myPermission' => 'myAssertion',
            'myPermission2' => [
                // multiple assertions
                'isAdmin',
                [
                    'isSharedWithEditPermission',
                    'isOwner',
                    // condition
                    'condition' => \ZfcRbac\Assertion\AssertionSet::CONDITION_AND,
                ],
                // condition
                'condition' => \ZfcRbac\Assertion\AssertionSet::CONDITION_OR,
            ],
        ],
    ],
];
````
at least I think this should work

Reply:
Nested assertions are already possible.

I noticed roles are checked first and when a particular role is allowed and assertions are defined these are evaluated. So in this example `isAdmin` really makes now sense (that is a role?).